### DISCUSSION TITLE
Is an empty boundaries array allowed?

### DISCUSSION THREAD

**Discussion by balazsdukai** [cityjson/specs#188] [Q&A] [answered]
Geometries are not obligatory for CityObjects, so we have three valid options:

- `{ "type": "Building" }`
- `{ "type": "Building", "geometries": [ ] }`
- `{ "type": "Building", "geometries": [ ... ] }`

Then, a `Geometry` object must have a `"boundaries"` member with an array of vertex indices. But the specs does not explicitly allow or forbid empty arrays, such as `"boundaries": [ ]`.

In my opinion, it does not make much sense to have a Geometry object with empty boundaries, but it can happen. For instance due to a failed procedure that creates the geometry.

Should we consider an empty `boundaries` a valid or invalid CityJSON?

**Comment by hugoledoux** [comment, marked as answer]
Good point, I had never thought of this! 

I think this can be fixed easily with adding a `minItems`, see that snippet: https://www.jsonschemavalidator.net/s/6GxMeDl8

**Reply by balazsdukai** [reply]
:+1:  this looks good to me, validation works indeed.
